### PR TITLE
draft, working on serialization

### PR DESCRIPTION
**Pull request by Minpeople**

### CODE REVIEW

File: app/controllers/cinema_halls_controller.rb [hunk 1]
```diff
@@ -1,9 +1,11 @@
 # frozen_string_literal: true
 
 class CinemaHallsController < ApplicationController
+include JSONAPI::Fetching
```
[the remark this answers is not in the document]
Comment (wmaciejak) [reply app/controllers/cinema_halls_controller.rb:4]: Of course if you want to rewrite all controllers to JSON:API it's fine, just thought that you created separated controllers under `/jsonapi` path :)

**Review by wmaciejak — CHANGES_REQUESTED** [review]
Except for one major comment  - `dependent` option, the rest looks really cool. Go through all endpoints and check if everything is consistent and please check if your tests passing. Good job 👏

File: Gemfile [hunk 2]
```diff
@@ -23,7 +23,9 @@ gem 'puma', '~> 5.0'
 # gem 'redis', '~> 4.0'
 # Use Active Model has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
-
+gem 'jsonapi.rb'
+gem 'jsonapi-serializer'
```
Comment (wmaciejak) [inline Gemfile:27]: It's a big chance that I misled you. We don't have to add `jsonapi-serializer'` cause it's already included in `jsonapi.rb` gem. Also remember to add version to `jsonapi.rb` gem

**Comment by Minpeople** [comment]
> Except for one major comment - `dependent` option, the rest looks really cool. Go through all endpoints and check if everything is consistent and please check if your tests passing. Good job 👏

Thank you!

File: app/controllers/screenings_controller.rb [hunk 3]
```diff
@@ -1,4 +1,10 @@
 # frozen_string_literal: true
 
 class ScreeningsController < ApplicationController
+  include JSONAPI::Fetching
+
+  def index 
+    screenings = Screenings.all
```
Comment (rafalrothenberger) [inline app/controllers/screenings_controller.rb:7]: Did you test that? Shouldn't this be `Screening.all` not `Screenings.all`?

File: Gemfile [hunk 4]
```diff
@@ -23,7 +23,8 @@ gem 'puma', '~> 5.0'
 # gem 'redis', '~> 4.0'
 # Use Active Model has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
-
+gem 'jsonapi.rb', '~> 1.7'
+gem 'pry', '~> 0.14.1'
```
Comment (rafalrothenberger) [inline Gemfile:27]: Wouldn't it be better to add it to `group :development, :test`?

File: app/controllers/screenings_controller.rb [hunk 5]
```diff
@@ -1,4 +1,10 @@
 # frozen_string_literal: true
 
 class ScreeningsController < ApplicationController
+  include JSONAPI::Fetching
+
+  def index
```
Comment (rafalrothenberger) [inline app/controllers/screenings_controller.rb:6]:
```suggestion
  def index
```

Did you run rubocop on your project?

File: Gemfile (hunk 4 above)
Comment (Minpeople) [reply Gemfile]: > Wouldn't it be better to add `pry` to `group :development, :test`?

@rafalrothenberger done :)

**Review by rafalrothenberger — APPROVED** [review]

**Pull request merged**